### ISSUE TITLE
Lab 5.1

### ISSUE THREAD

**Issue by trickypepe** [nus-cs2030/2223-s2#359]
# Summary

Hi, looking for suggestions as to why the serve time is right but is not finished on the right time. 
(yep its that test case)

## Screenshots
![Screenshot 2023-03-11 145153](https://user-images.githubusercontent.com/122513341/224469901-3fadcb5a-288a-448f-9f45-ac5ffb225afb.png)

**Comment by Maxxtucker** [comment]
I also encounter this problem. They seem to change the Main.java for the test case and I don't know how to solve it.

**Comment by trickypepe** [comment]
> I also encounter this problem. They seem to change the Main.java for the test case and I don't know how to solve it.

Hi, I just solved the issue. The problem occurs when server goes on break while there is a queue. I "scheduled" the server to serve the next customer and removed from the queue immediately. This is wrong and instead of resolving it immediately I had to return a new Served event (my implementation, yours may differ).

Basically I was not supposed to remove from the queue immediately after 1 serve event. Hope it helps!
